### ISSUE TITLE
Possible regression bug: null property as variable going from 3.3 to 3.4

### ISSUE THREAD

**Issue by mdfrenchman** [neo4j/neo4j#12000]
Tested this out on both 3.3 and 3.4 and was able to reproduce trimming it down to the below error.

- **Neo4j version:** 3.4.0
- **Operating system:** WIndows 10
- **API/Driver:** C# driver, and Cypher via browser
- **Steps to reproduce**
Setup Cypher:
    `MERGE (:Unit)-[:RELATED]->(:Detail {StartDate: 20171001, EndDate: 20180101})`
Error Occurs here: 
```
// StationCode does not exist as a property on ua. 
// It doesn't trigger the error until the MERGE statement is added.
MATCH (u:Unit)-[:RELATED]->(ua)
MATCH (s:Station {StationCode: ua.StationCode})
MERGE (u)-[:STATIONED_AT]->(s) 
```
- **Expected behavior**
On version 3.3.0 it will return, having done nothing
- **Actual behavior**
On version 3.4.0 it will throw an error.
Error: Neo.DatabaseError.General.UnknownError

**Comment by SaschaPeukert** [comment]
@mdfrenchman Thanks for reporting. I just tried it in `Neo4j 3.4.5` and it seems to be working there. Could you update to that and test your query with that version again?

**Comment by sgehrig** [comment]
Perhaps related to #11842? This has been fixed in 3.4.1.

**Comment by mdfrenchman** [comment]
Very possible. I’ll upgrade and verify when I’m back to a computer.

**Comment by mdfrenchman** [comment]
Confirm this is fixed in 3.4.5
